heartbeat/monitors/wrappers/monitorstate/tracker_test.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package monitorstate
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrackerRecord(t *testing.T) {
+	mst := NewTracker(NilStateLoader, true)
+	ms := mst.RecordStatus(TestSf, StatusUp)
+	require.Equal(t, StatusUp, ms.Status)
+	requireMSStatusCount(t, ms, StatusUp, 1)
+
+	for i := 0; i < FlappingThreshold; i++ {
+		_ = mst.RecordStatus(TestSf, StatusDown)
+		ms = mst.RecordStatus(TestSf, StatusUp)
+	}
+	require.Equal(t, StatusFlapping, ms.Status)
+	requireMSCounts(t, ms, FlappingThreshold+1, FlappingThreshold)
+
+	// Restore stable state
+	for i := 0; i < FlappingThreshold; i++ {
```

Comment:
nit: use recordStableSeries?

Reply:
It's in another package. It's quite irritating that you can't use stuff from other package's `_test.go` files without extra complexity. Given that this is just here, I think it's simplest to just have the loop here :(

Sometimes I just add the test helpers to a `testhelpers.go` file in that package to expose them, but it always feels a bit hacky.